ERCS/erc-7902.md

```diff
@@ -0,0 +1,206 @@
+the Wallet Application to resolve any dynamic configuration.
+
+The application may hard-code or resolve these parameters first and pass them with this capability.
+
+Identifier:
+
+`staticPaymasterConfiguration`
+
+Interface:
+
+```typescript
+type StaticPaymasterConfigurationCapabilityParams = Record<
+  {
+    paymaster: string;
+    paymasterData: string;
+    paymasterValidationGasLimit: `0x${string}`;
+    paymasterPostOpGasLimit: `0x${string}`;
+  }
+>;
+```
+
+### Validity Time Range Capability
+
+The purpose of this capability is allowing the applications to explicitly specify the time range during which
+the requested operations will be valid after signing.
+
+Identifier:
+
+`validityTimeRange`
+
+Interface:
+
+```typescript
+type ValidityTimeRangeCapabilityParams = Record<
+  {
+    validAfter: `0x${string}`, // operation valid only after this timestamp, in seconds
+    validUntil: `0x${string}`  // operation valid only before this timestamp, in seconds
+  }
+>
+```
+
+The Wallet Application MUST verify the time range [`validAfter`..`validUntil`] is valid and present it to the
+user in a human-readable way for confirmation as part of the transaction information.
+
+The Smart Contract Account MUST specify the time range [`validAfter`..`validUntil`] as the transaction validity range.
+
+### Multidimensional Nonce Capability
+
+The purpose of this capability is allowing the applications to explicitly specify the components of the
+semi-abstracted nonce as defined in [ERC-4337](./eip-4337.md).
+
+Identifier:
+
+`multiDimensionalNonce`
+
+Interface:
+
+```typescript
+type MultiDimensionalNonceCapabilityParams = Record<
+  {
+    nonceKey: `0x${string}`,
+    nonceSequence: `0x${string}`
+  }
+>
+```
+
+For Smart Contract Accounts that support multidimensional nonce values,
+the wallet must specify these parameters during the actual on-chain execution of the batch.
+
+### Account Abstraction Gas Parameters Override Capability
+
+The purpose of this capability is allowing the applications to override the Wallet Application's suggested values
+for all gas-related parameters.
+
+This capability provides very low-level access to the underlying Account Abstraction protocol and should only
+be used by applications closely coupled to a specific version of a specific protocol.
+It may also prove useful in the context of development and debugging.
+
+It is generally recommended that production dapps rely on higher-level features of Wallet Applications instead.
+
+Identifier:
+
+`accountAbstractionGasParamsOverride`
+
+Interface:
+
+```typescript
+type AAGasParamsOverrideCapabilityParams = Record<
+  {
+    preVerificationGas?: `0x${string}`,
+    verificationGasLimit?: `0x${string}`,
+    callGasLimit?: `0x${string}`,
+    paymasterVerificationGasLimit?: `0x${string}`,
+    paymasterPostOpGasLimit?: `0x${string}`,
+    maxFeePerGas?: `0x${string}`,
+    maxPriorityFeePerGas?: `0x${string}`
```

Comment:
one usecase we've seen as well has been to be able to request a scaling factor override. might be useful to support that here as well ie
```
preVerificationGas: `0x${string}` | { mult: number }
```

as an example, scaling these values by 50% would look like:
```
preVerificationGas: { mult: 1.5 }
```

Reply:
Hello @moldy530 , sorry for a late reply. When providing the `mult` value to the `preVerificationGas`, what would be the base value to which the factor is applied?

Reply:
all good! 

the base value in our case has been w/e the bundler or paymaster service estimate these values to be